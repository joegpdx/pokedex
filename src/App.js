import React, { Component } from 'react';
import request from 'superagent';
import Header from './Header.js';
import PokeList from './PokeList.js';
import './App.css';


export default class App extends Component {
  state = {
    pokemon: [],
    query: ''
  }


  async componentDidMount() {
    const query = this.state.query;
    const data = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex`)

    console.log(data.body);

    this.setState({ pokemon: data.body.results })
  }

  render() {
    return (
      <div>
        <Header />
        <PokeList pokemons={this.state.pokemon} />
      </div>
    );
  }
}