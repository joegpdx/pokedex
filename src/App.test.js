import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('renders App.js correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
