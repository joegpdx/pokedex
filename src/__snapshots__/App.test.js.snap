// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App.js correctly 1`] = `
<div>
  <header>
    <h1>
      Pokemon Cards
    </h1>
  </header>
  <div
    id="search"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        value=""
      />
      <button>
        Search
      </button>
    </form>
  </div>
  <ul />
  <p
    className="paging"
  >
    No results, try another search
  </p>
</div>
`;
